server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -706,6 +700,15 @@ protected void writeIndexGen(final RepositoryData repositoryData, final long rep
         }
         logger.debug("Repository [{}] updating index.latest with generation [{}]", metadata.name(), newGen);
         writeAtomic(INDEX_LATEST_BLOB, genBytes, false);
+        // delete the N-2 index file if it exists, keep the previous one around as a backup
+        if (isReadOnly() == false && newGen - 2 >= 0) {
```

Comment:
let's remove this `isReadOnly()` check - it's bogus and we earlier assert already that this method is never called on a readonly repo.